src/System.Management.Automation/engine/runtime/Binding/Binders.cs

```diff
@@ -2940,13 +2940,15 @@ private DynamicMetaObject BinaryBitwiseAnd(DynamicMetaObject target, DynamicMeta
 
                 if (numericTarget != null && numericArg != null)
                 {
-                    var expr = exprGenerator(numericTarget.Expression.Cast(target.LimitType).Cast(opType),
-                        numericArg.Expression.Cast(numericArg.LimitType).Cast(opType)).Cast(typeof(object));
+                    var expr = exprGenerator(numericTarget.Expression.Cast(numericTarget.LimitType).Cast(opType),
+                                             numericArg.Expression.Cast(numericArg.LimitType).Cast(opType));
 
                     if (target.LimitType.GetTypeInfo().IsEnum)
                     {
-                        expr = expr.Cast(target.LimitType).Cast(typeof(object));
+                        expr = expr.Cast(target.LimitType);
                     }
+
+                    expr = expr.Cast(typeof(object));
```

Comment:
`.Cast(typeof(object))` is removed from `var expr = exprGenerator(...)`. In case the target is not a enum, the `if (target.LimitType.GetTypeInfo().IsEnum)` block will not run, but we still want the final expression to be converted to `System.Object`. That's why I removed the cast outside the IsEnum block.